### PR TITLE
WIP: stdcall name mangling

### PR DESCRIPTION
**Pull request by vtjnash**
ref #7375

### CODE REVIEW

File: src/ccall.cpp [hunk 1]
```diff
@@ -898,6 +898,17 @@ static Value *emit_ccall(jl_value_t **args, size_t nargs, jl_codectx_t *ctx)
...
 
         PointerType *funcptype = PointerType::get(functype,0);
+        
+        SmallString<20> f_name_mangled;
+        if (1) {
+            const Function *fn = Function::Create(functype, GlobalValue::ExternalLinkage, f_name, NULL);
+            jl_mang->getNameWithPrefix(f_name_mangled, fn, false);
+            //printf("%s -> %s\n", f_name, (char*)f_name_mangled.c_str());
+            f_name = (char*)f_name_mangled.c_str();
+            f_name += strlen(jl_Ctx->getAsmInfo().getGlobalPrefix()); // strip leading '_', etc.
```
Comment (Keno) [inline src/ccall.cpp]: Yeah, I don't know either. If I had to guess this is probably fine, but I can't say.

File: src/ccall.cpp [hunk 2]
```diff
@@ -898,6 +898,17 @@ static Value *emit_ccall(jl_value_t **args, size_t nargs, jl_codectx_t *ctx)
         assert(f_name != NULL);
 
         PointerType *funcptype = PointerType::get(functype,0);
+        
+        SmallString<20> f_name_mangled;
+        if (1) {
+            const Function *fn = Function::Create(functype, GlobalValue::ExternalLinkage, f_name, NULL);
+            jl_mang->getNameWithPrefix(f_name_mangled, fn, false);
```
Comment (Keno) [inline src/ccall.cpp]: I suspect that might be right.

**Comment by Keno** [comment]
Should we add a separate `winapi` calling convention to `ccall` that does not go through the Mangler?

**Comment by Keno** [comment]
Also needs a rebase.

**Comment by tkelman** [comment]
> Should we add a separate `winapi` calling convention to `ccall` that does not go through the Mangler?

Sounds like a good idea to me.

**Comment by MrMormon** [comment]
?

**Comment by vtjnash** [comment]
https://github.com/JuliaLang/julia/issues/7375#issuecomment-47428220 is better

**Pull request closed without merging**